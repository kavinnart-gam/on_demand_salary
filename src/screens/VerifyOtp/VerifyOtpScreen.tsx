import React from 'react';
import {StyleSheet, View, Text} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import OtpInputs from 'react-native-otp-inputs';
import {asyncStorage, common} from '../../utils';
import {useAuth} from '../../context/AuthContext';

function VerifyOtpScreen({navigation}: any): React.JSX.Element {
  const {onUpdateAuth} = useAuth();

  const handleChange = async (otpCode: string) => {
    const pinCode = await asyncStorage.getDataFromAsyncStorage({
      key: 'pincode',
    });

    if (otpCode.length === 6) {
      setTimeout(() => {
        onUpdateAuth!();
        if (pinCode) {
          navigation.navigate('Bottomtab');
        } else {
          navigation.navigate('PassCode');
        }
      }, 100);
    }
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.contentHeader} {...common.testID('TXT_VERIFY_OTP')}>
        <Text style={styles.title}>Verify OTP</Text>
      </View>
      <View {...common.testID('INPUT_TXT_OTP')}>
        <OtpInputs
          handleChange={code => {
            handleChange(code);
          }}
          numberOfInputs={6}
          autofillFromClipboard={false}
          keyboardType="phone-pad"
          returnKeyLabel="done"
          inputStyles={styles.otpInput}
        />
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#debbb0',
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentHeader: {
    marginBottom: 20,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  otpInput: {
    height: 70,
    width: 50,
    margin: 5,
    borderWidth: 1,
    borderColor: '#000',
    color: 'black',
    borderRadius: 12,
    backgroundColor: 'white',
    fontSize: 24,
    textAlign: 'center',
  },
});

export default VerifyOtpScreen;
