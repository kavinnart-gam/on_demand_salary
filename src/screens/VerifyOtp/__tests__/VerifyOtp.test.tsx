import {render} from '@testing-library/react-native';
import VerifyOtpScreen from '../VerifyOtpScreen';

describe('VerifyOtp', () => {
  it('Verification screen is match snapshot', () => {
    const result = render(<VerifyOtpScreen />);
    expect(result).toMatchSnapshot();
  });

  it('VerifyOtpScreen component should be render correctly', () => {
    const {getByTestId} = render(<VerifyOtpScreen />);
    const inputOtp = getByTestId('INPUT_TXT_OTP');
    const textOtp = getByTestId('TXT_VERIFY_OTP');
    expect(inputOtp).toBeTruthy();
    expect(textOtp).toBeTruthy();
  });
});
