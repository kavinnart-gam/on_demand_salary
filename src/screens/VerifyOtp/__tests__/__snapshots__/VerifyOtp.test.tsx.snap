// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyOtp Verification screen is match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#debbb0",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityLabel="TXT_VERIFY_OTP"
    accessible={true}
    style={
      {
        "marginBottom": 20,
      }
    }
    testID="TXT_VERIFY_OTP"
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Verify OTP
    </Text>
  </View>
  <View
    accessibilityLabel="INPUT_TXT_OTP"
    accessible={true}
    testID="INPUT_TXT_OTP"
  />
</RNCSafeAreaView>
`;
